python/src/gen_gherkin/generate_gherkin_feature_files_impl.py

```diff
@@ -4,43 +4,273 @@
+                    scenario_lists.append(collect_and_sanitize_scenario_steps(scenario))
+    if "requirements" in behavior_entry:
+        for requirement in behavior_entry["requirements"]:
+            behavior_requirements.append(requirement)
+
+    return [
+        {
+            "feature": {"name": feature_name, "description": feature_description},
+            "scenarios": [scenario for scenario_list in scenario_lists for scenario in scenario_list],
+            "behavior_requirements": behavior_requirements,
+        }
+    ]
+
+
+def collect_model_behavior_properties(model: Definition) -> dict:
+    """
+    Produce a template property dictionary for each behavior entry in a model.
+
+    Args:
+        model (Definition): A model containing behavior properties
+
+    Returns:
+        A dictionary containing a list of behaviors and a list of requirements
+    """
+    behaviors = []
+    requirements = []
+    behavior_lists = []
+    if "behavior" in model.content:
+        for behavior in model.structure["model"]["behavior"]:
+            behaviors.append(behavior)
+            if "requirements" in behavior:
+                for requirement in behavior["requirements"]:
+                    requirements.append(requirement)
+    for behavior in behaviors:
+        behavior_lists.append(collect_behavior_entry_properties(behavior))
+    returning_list = {
+        "name": model.name,
+        "behaviors": [behavior for behavior_list in behavior_lists for behavior in behavior_list],
+        "model_requirements": requirements
+    }
+
+    return returning_list
+
+
+def does_step_start_with_gherkin_keyword(step: str) -> bool:
+    """
+    Check if a string starts with a Gherkin keyword. Gherkin keywords can be found here: https://cucumber.io/docs/gherkin/reference/#keywords.
+
+    Args:
+        step (str): The scenario step being checked
+
+    Returns:
+        A boolean value signifying whether the step begins with a gherkin keyword.
+    """
+    gherkin_keywords = [
+        "Feature",
+        "Rule",
+        "Example",
+        "Given",
+        "When",
+        "Then",
+        "And",
+        "But",
+        "Background",
+        "Example",
+        "Scenario",
+        "Scenario Outline",
+        "Scenario Template",
+    ]
+
+    return step.startswith(tuple(gherkin_keywords))
+
+
+def _get_template_properties(parsed_models: dict) -> list[dict]:
+    """
+    Generate a list of template property dictionaries for each gherkin feature file to generate.
+
+    Args:
+        parsed_models (dict): a dict of models where the key is the model name and the value is the model dict
+
+    Returns:
+        a list of template property dictionaries
+    """
+
+    return [collect_model_behavior_properties(model) for model in collect_models(parsed_models)]
+
+
+def after_gen_gherkin_behaviors(architecture_file: str, output_directory: str, run_generate: Callable) -> ExecutionResult:
```

Comment:
So, this method should not be this populated. Generate has the execution status handling as part of its logic. You should just be calling generate. Take a look at the PUML plugin or gen_plugin within core for an example.

Reply:
I think it is the multiple runs of generate that causes me to have to do this.   That error I messaged you about yesterday, where the messages were not bubbling up properly, was caused by this.